quantized_mesh_encoder/encode.py

```diff
@@ -70,7 +77,18 @@ def encode(
         ellipsoid,
         Ellipsoid), ('ellipsoid must be an instance of the Ellipsoid class')
 
-    header = compute_header(positions, sphere_method, ellipsoid=ellipsoid)
+    # TODO: Also check no duplicate instances of an extension type
+    assert all(
+        isinstance(ext, ExtensionBase) for ext in extensions
+    ), 'extensions must be list of instances of the Extension class'
+
+    cartesian_positions = to_ecef(positions, ellipsoid=ellipsoid)
+
+    header = compute_header(
+        positions,
+        np.copy(cartesian_positions),
```

Comment:
Why do we need a copy here?

Reply:
For some reason `cartesian_positions`  was getting mutated somewhere in the `computer_header()` function.
Left a little note on this and other likely ill-advised changes I made [here](https://github.com/kylebarron/quantized-mesh-encoder/pull/35#discussion_r597322291).